### PR TITLE
Add `aria-label` and/or `title` to `<button>` elements

### PR DESCRIPTION
**Pull request by lyzadanger**
Hopefully, third time's the charm here, and with luck:

Fixes: https://github.com/hypothesis/client/issues/2056

This PR adds `aria-label` attributes to `<button>` elements when their title/label differs from the displayed button text or if there is no button text.

Depends on #2130

### CODE REVIEW

File: src/sidebar/components/excerpt.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ function InlineControls({ isCollapsed, setCollapsed, linkStyle = {} }) {
         <button
           className="excerpt__toggle-button"
           onClick={() => setCollapsed(!isCollapsed)}
-          aria-label="Toggle to show the full excerpt or first few lines only"
+          aria-expanded={!isCollapsed}
+          aria-label="Toggle to show the full excerpt"
           aria-pressed={(!isCollapsed).toString()}
```
Comment (LMS007) [inline src/sidebar/components/excerpt.js:25]: While we're touching this, I think can convert a few of these over. I think

`aria-pressed={!isCollapsed}`

https://github.com/preactjs/preact/pull/2347

File: src/sidebar/components/moderation-banner.js [hunk 2]
```diff
@@ -51,15 +51,16 @@ function ModerationBanner({ annotation, api, toastMessenger }) {
   };
 
   const toggleButtonProps = (() => {
```
Comment (LMS007) [inline src/sidebar/components/moderation-banner.js:53]: This is toggle button but we don't have an `aria-pressed` prop.  I suppose thats okay when we change the title like this. But I see we're not changing the title when using `aria-expanded` (in AnnotationBody) Should we also apply that standard for aria-pressed? I'm a bit confused by that.

Comment (lyzadanger) [reply src/sidebar/components/moderation-banner.js:53]: Can we please put non-label/title things aside for this set of changes?

File: src/sidebar/components/excerpt.js (hunk 1 above)
Comment (lyzadanger) [reply src/sidebar/components/excerpt.js:25]: Feel free to do so! I really want to focus just on labels and titles in this PR.

**Pull request merged**